Kernel/Net/LocalSocket.cpp

```diff
@@ -399,14 +399,27 @@ ErrorOr<void> LocalSocket::getsockopt(OpenFileDescription& description, int leve
 
     MutexLocker locker(mutex());
 
+    size_t buffer_size;
     socklen_t size;
     TRY(copy_from_user(&size, value_size.unsafe_userspace_ptr()));
 
     switch (option) {
     case SO_SNDBUF:
-        return ENOTSUP;
+        if (size < sizeof(size_t))
+            return EINVAL;
+        buffer_size = m_for_server.ptr()->space_for_writing();
+        TRY(copy_to_user(static_ptr_cast<int*>(value), bit_cast<int*>(&buffer_size)));
+        size = sizeof(size_t);
+        TRY(copy_to_user(value_size, &size));
+        return {};
     case SO_RCVBUF:
-        return ENOTSUP;
+        if (size < sizeof(size_t))
```

Comment:
(GitHub doesn't let me select multiple lines for review right now)

Same corrections from above apply to SO_RCVBUF as well.